tests/usm/usm_atomic_access.h

```diff
@@ -0,0 +1,173 @@
+/*******************************************************************************
+//
+//  SYCL 2020 Conformance Test Suite
+//
+//  Provides common code for USM atomic access tests
+//
+*******************************************************************************/
+
+#ifndef __SYCL_CTS_TEST_USM_ATOMIC_ACCESS_H
+#define __SYCL_CTS_TEST_USM_ATOMIC_ACCESS_H
+
+#include "../../util/usm_helper.h"
+#include "../common/type_coverage.h"
+#include <chrono>
+#include <cstring>
+#include <limits.h>
+#include <thread>
+#include <type_traits>
+
+namespace usm_atomic_access {
+
+static auto get_scalar_types() {
+  static const auto scalar_types =
+      named_type_pack<short, unsigned short, int, unsigned int, long,
+                      unsigned long, float, double, long long,
+                      unsigned long long>({"short", "unsigned short", "int",
+                                           "unsigned int", "long",
+                                           "unsigned long", "float", "double",
+                                           "long long", "unsigned long long"});
+  return scalar_types;
+}
+
+/** @brief Return atomic aspect depending on the type of allocated memory
+ *  @retval AllocMemT USM allocation type
+ */
+template <sycl::usm::alloc AllocMemT>
+auto get_atomic_memory() {
+  if constexpr (AllocMemT == sycl::usm::alloc::host) {
+    return sycl::aspect::usm_atomic_host_allocations;
+  } else if constexpr (AllocMemT == sycl::usm::alloc::shared) {
+    return sycl::aspect::usm_atomic_shared_allocations;
+  } else {
+    static_assert(AllocMemT != AllocMemT,
+                  "This USM allocation type does not supported");
+  }
+}
+
+using with_atomic64 = std::true_type;
+using without_atomic64 = std::false_type;
+/** @brief Dummy functor that use in type coverage
+ *  @tparam CounterT Variable type from type coverage
+ *  @tparam UseAtomic64Flag std::integral_constant type
+ */
+template <typename CounterT, typename UseAtomic64Flag>
+struct run_all_tests {
+  void operator()(sycl::queue &queue, sycl_cts::util::logger &log,
+                  const std::string &typeName) {
+    if constexpr ((sizeof(CounterT) * CHAR_BIT) == 64) {
+      if (!UseAtomic64Flag::value) {
```

Comment:
```suggestion
      if constexpr (!UseAtomic64Flag::value) {
```

If I understand this correctly

Reply:
I've applied this comment in this commit: 4e8f956ce1f4490d5d5f2379f0f26d7342c8376c